app/uk/gov/hmrc/helptosave/controllers/EligibilityCheckController.scala

```diff
@@ -31,8 +31,8 @@ class EligibilityCheckController @Inject() (val helpToSaveService: HelpToSaveSer
     transformer: LogMessageTransformer, appConfig: AppConfig, ec: ExecutionContext)
   extends HelpToSaveAuth(authConnector) with EligibilityBase {
 
-  def eligibilityCheck(): Action[AnyContent] = ggAuthorisedWithNino { implicit request ⇒ implicit nino ⇒
-    checkEligibility(nino, routes.EligibilityCheckController.eligibilityCheck().url)
+  def eligibilityCheck(nino: Option[String]): Action[AnyContent] = ggOrPrivilegedAuthorisedWithNINO(nino) { implicit request ⇒ implicit authNino ⇒
+    checkEligibility(authNino, routes.EligibilityCheckController.eligibilityCheck(Some(authNino)).url)
```

Comment:
Yes, thank you

Reply:
This causes a unit test to fail and I'm unsure how to get it to pass as the NINO is required in a lot of places instead of being optional, i.e. HelpToSaveService.getEligibility